src/components/_product/HeroPhysical/HeroPhysical.tsx

```diff
@@ -0,0 +1,217 @@
+            value: [a.value],
+          });
+        } else if (!arr[index]!.value.includes(a.value)) {
+          if (arr[index]!.value) {
+            arr[index]!.value.push(a.value);
+          }
+        }
+      });
+    });
+
+    return arr;
+  }, [variants]);
+
+  const [count, setCount] = useState(1);
+  const [chosenVariant, setChosenVariant] = useState(variants ? variants[0] : physical);
+  const [selectedImage, setSelectedImage] = useState(0);
+  const [chosenAttributes, setChosenAttributes] = useState(() => {
+    const obj = {} as SelectedAttributesTypes;
+    attributes.forEach((el) => {
+      obj[el.name] = el.value[0];
+    });
+    return obj;
+  });
+
+  const images = useMemo(() => {
+    const images = [];
+    // add video as first element if exists
+    if (chosenVariant?.featuredVideo) images.push({ type: 'video', data: chosenVariant?.featuredVideo });
+    chosenVariant?.gallery!.forEach((el) => images.push({ type: 'image', data: el }));
+    return images;
+  }, [chosenVariant]);
+
+  const selectVariant = useCallback(
+    (v: SingleValue<{ value: string | undefined; label: string | undefined }>, name: string) => {
+      const filteredAttributes = { ...chosenAttributes };
+      filteredAttributes[name] = v?.value;
+
+      // find variant with all the same attributes
+      const filteredVariant = variants.find((el) => {
+        let all = true;
+        el.attributes!.every((attr) => {
+          if (filteredAttributes[attr.name.toLocaleLowerCase()] !== attr.value) all = false;
+
+          return all;
+        });
+        return all;
+      });
+
+      setChosenVariant(filteredVariant);
+      setChosenAttributes(filteredAttributes);
+      setSelectedImage(0);
+      setCount(filteredVariant!.countInStock > 0 ? 1 : 0);
+    },
+    [variants, chosenAttributes]
+  );
+
+  return (
+    <section className={styles['HeroPhysical']}>
+      <div className={styles['gallery']}>
+        {gallerySwitch(images[selectedImage]!.data)[images[selectedImage]!.type as 'image' | 'video']}
+        <div className={styles['gallery-grid']}>
+          {images.map((el, index) => {
+            if (index === selectedImage) return null;
+            return (
+              <button
+                onClick={() => setSelectedImage(index)}
+                key={index}
+              >
+                {gallerySwitch(el.data)[el.type as 'image' | 'video']}
+              </button>
+            );
+          })}
+        </div>
+        {/* TODO: add arrows */}
+      </div>
+      <div className={styles['info']}>
+        {/* <p>reviews</p> TODO: add reviews */}
+        <h1>{name}</h1>
+        <div className={styles.attributes}>
+          {attributes.map((el) => (
+            <p key={el.name}>
+              <span>Wybierz {el.name}</span>
+              <Select
+                defaultValue={{ value: chosenAttributes[el.name], label: chosenAttributes[el.name] }}
+                onChange={(e) => selectVariant(e, el.name)}
```

Comment:
While changing the variant, I would use URL params or path, to track the picked variant. Potentially will be beneficial while users will want to share specific variants.

Reply:
This type of logic will cause page to be SSR and cause high decrease od page speed.